doc/dev/adr/0023-property-support.md

```diff
@@ -0,0 +1,306 @@
+* WriteError defines the type returned by the Write API when an error occurs during writing.
+
+In addition, the schema of a Field that is co-typed ReadError or WriteError must be an Object that is co-typed with the extant Error adjunct type.
+
+The following sample model illustrates the use of these new adjunct types in an enhancement of the above model.
+
+```json
+{
+  "@context": [ "dtmi:dtdl:context;4", "dtmi:dtdl:extension:mqtt;4" ],
+  "@id": "dtmi:propertySketch:PropertySketch;1",
+  "@type": [ "Interface", "Mqtt" ],
+  "payloadFormat": "Json/ecma/404",
+  "propertyTopic": "sample/property/{propertyName}/{action}",
+  "contents": [
+    {
+      "@type": "Property",
+      "name": "Foo",
+      "schema": {
+        "@type": [ "Object", "PropertyResult" ],
+        "fields": [
+          {
+            "@type": [ "Field", "PropertyValue" ],
+            "name": "foo",
+            "schema": "integer"
+          },
+          {
+            "@type": [ "Field", "ReadError" ],
+            "name": "propError",
+            "schema": "dtmi:com:example:FooPropertyError;1"
+          }
+        ]
+      }
+    },
+    {
+      "@type": "Property",
+      "name": "Bar",
+      "writable": true,
+      "schema": {
+        "@type": [ "Object", "PropertyResult" ],
+        "fields": [
+          {
+            "@type": [ "Field", "PropertyValue" ],
+            "name": "bar",
+            "schema": "string"
+          },
+          {
+            "@type": [ "Field", "ReadError", "WriteError" ],
+            "name": "propError",
+            "schema": "dtmi:com:example:BarPropertyError;1"
+          }
+        ]
+      }
+    }
+  ]
+}
+```
+
+Note that the "propError" field of Property "Bar" is co-typed with both ReadError and WriteError.
+This is not a requirement.
+Writable Properties may define separate error result types for read and write Commands if desired.
+
+### New adjunct type: Fragmented
+
+The third additive change to the Mqtt extension is support for dynamically itemized Properties via the new adjunct type Fragmented.
+
+The following sample model defines a dynamically itemized, writable Property placeholder named Baz.
+
+```json
+{
+  "@context": [ "dtmi:dtdl:context;4", "dtmi:dtdl:extension:mqtt;4" ],
+  "@id": "dtmi:propertySketch:PropertySketch;1",
+  "@type": [ "Interface", "Mqtt" ],
+  "payloadFormat": "Json/ecma/404",
+  "propertyTopic": "sample/property/{propertyName}/{action}",
+  "contents": [
+    {
+      "@type": [ "Property", "Fragmented" ],
+      "name": "Baz",
+      "writable": true,
+      "schema": {
+        "@type": "Map",
+        "mapKey": {
+          "name": "propKey",
+          "schema": "string"
+        },
+        "mapValue": {
+          "name": "propValue",
+          "schema": "integer"
+        }
+      }
+    }
+  ]
+}
+```
+
+A dynamically itemized Property placeholder is defined using co-type Fragmented and a schema that is a Map whose contents are dynamically added Properties.
```

Comment:
is the `Fragmented` co-type needed, or can we infer this if the schema is a Map? Or does `Fragmented` allow for a different scenario that doesn't include `Fragmented` and treats the Map as any other field that gets fully defined/replaced per write?

Reply:
I think it is needed; the explanation of the semantics is in the subsequent paragraph.  And yes, this is to distinguish from cases in which the user models a normal static Property that happens to have a Map for its schema.